components/prize_pool/commons/prize_pool_import.lua

```diff
@@ -0,0 +1,513 @@
+-- third place match to determine placement.
+function Import._getPreTiebreakMatchIds(bracket)
+	local firstBracketData = bracket.bracketDatasById[bracket.rootMatchIds[1]]
+	local thirdPlaceMatchId = firstBracketData.thirdPlaceMatchId
+
+	local sfMatchIds = {}
+	if thirdPlaceMatchId and bracket.matchesById[thirdPlaceMatchId] then
+		for _, lowerMatchId in ipairs(firstBracketData.lowerMatchIds) do
+			sfMatchIds[lowerMatchId] = true
+		end
+	end
+	return sfMatchIds
+end
+
+-- Finds the first round in where upper bracket opponents drop to the lower
+-- bracket. Returns nil if it cannot be determined unambiguously, or if the
+-- bracket is not double elimination.
+function Import._findDeRoundIndex(bracket)
+	if #bracket.sections ~= 2 then
+		return
+	end
+	local countsByRound = MatchGroupCoordinates.computeRawCounts(bracket)
+
+	for roundIndex = 1, #bracket.rounds do
+		local lbCount = countsByRound[roundIndex][2]
+		if lbCount == 0 then
+			return roundIndex
+		elseif lbCount > 0 then
+			return
+		end
+	end
+end
+
+function Import._mergePlacements(lpdbEntries, placements)
+	for placementIndex, lpdbPlacement in ipairs(lpdbEntries) do
+		placements[placementIndex] = Import._mergePlacement(
+			lpdbPlacement,
+			placements[placementIndex] or Import._emptyPlacement(placements[placementIndex - 1], #lpdbPlacement)
+		)
+	end
+
+	return placements
+end
+
+function Import._emptyPlacement(priorPlacement, placementSize)
+	priorPlacement = priorPlacement or {}
+	local placeStart = (priorPlacement.placeEnd or 0) + 1
+	local placeEnd = (priorPlacement.placeEnd or 0) + placementSize
+
+	local parent = Table.deepCopy(priorPlacement.parent, {copyMetatable = true})
+	parent.placements = nil
+
+	return Placement(
+		{placeStart = placeStart, placeEnd = placeEnd},
+		parent,
+		priorPlacement.placeEnd or 0
+	)
+end
+
+function Import._mergePlacement(lpdbEntries, placement)
+	for opponentIndex, opponent in ipairs(lpdbEntries) do
+		placement.opponents[opponentIndex] = Import._mergeEntry(
+			opponent,
+			Table.mergeInto(placement:_parseOpponents{{}}[1], placement.opponents[opponentIndex]),
+			placement
+		)
+	end
+
+	assert(
+		#placement.opponents <= 1 + placement.placeEnd - placement.placeStart,
+		'Import: Too many opponents returned from query for placement range '
+			.. placement:_displayPlace():gsub('&#045;', '-')
+	)
+
+	return placement
+end
+
+function Import._mergeEntry(lpdbEntry, entry, placement)
+	if
+		not Opponent.isTbd(entry.opponentData) -- valid manual input
+		or Table.isEmpty(lpdbEntry.opponent) -- irrelevant lpdbEntry
+		or Opponent.isTbd(lpdbEntry.opponent)
+	then
+		return entry
+	end
+
+	entry.opponentData = Import._removeTbdIdentifiers(entry.opponentData)
+	return Table.deepMergeInto(Import._entryToOpponent(lpdbEntry, placement), entry)
+end
+
+function Import._removeTbdIdentifiers(opponent)
+	if Table.isEmpty(opponent) then
+		return {}
+	elseif not Opponent.isTbd(opponent) then
+		opponent.isAlreadyParsed = true
```

Comment:
Would it make more sense to have it outside as a `if Table.empty() then parsed = true`?

Reply:
`isNotEmpty`!!!

but yeah could do it in the `Import._entryToOpponent` function